### PR TITLE
Auth backend integration

### PR DESCRIPTION
**Pull request by arturjaworski**
Resolving #3

### CODE REVIEW

File: Customer Demo/Customer Demo/Screens/Setup/SetupPresenter.swift [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    weak var viewController: SetupViewController?
+    
+    func handleSubmitAction(
+        sourceLocationText: (String, String),
+        destinationLocationText: (String, String),
+        minimumDisplacementText: String,
+        desiredIntervalText: String,
+        accuracyIndex: Int
+    ) {
+        if !validate(
```
Comment (JakubJankowski) [inline Customer Demo/Customer Demo/Screens/Setup/SetupPresenter.swift:23]: `guard` here would be probably more readable, but it's such a minor thing that you can feel free to ignore me...

File: Customer Demo/Customer Demo/Screens/Setup/SetupPresenter.swift [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            )
+
+            self?.viewController?.navigationController?.pushViewController(deliveryStatusViewController, animated: true)
+        }
+    }
+}
+
+private extension SetupPresenter {
+    
+    func createOrder(source: CLLocation, destination: CLLocation, _ completion: @escaping (OrderResponse) -> Void) {
```
Comment (JakubJankowski) [inline Customer Demo/Customer Demo/Screens/Setup/SetupPresenter.swift:72]: Hmm, maybe we should move the networking logic to a separate class/service? I'm not a biggest fan of putting it into a presenter.

File: Customer Demo/Customer Demo/Screens/SignIn/SignInViewController.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
+//
+//  SignInViewController.swift
+//  Customer Demo
+//
+//  Copyright 2022 Ably Real-time Ltd (ably.com)
+//
+
+import Foundation
```
Comment (JakubJankowski) [inline Customer Demo/Customer Demo/Screens/SignIn/SignInViewController.swift]: `Foundation` import is redundant if you're already importing `UIKit`, can be removed

File: Customer Demo/Customer Demo/Screens/SignIn/SignInViewController.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    @IBOutlet private var submitButton: UIButton!
+    
+    let presenter = SignInPresenter()
+    
+    override func viewDidLoad() {
+        presenter.viewController = self
+        
+        super.viewDidLoad()
+        
+        submitButton.addTarget(self, action: #selector(submitButtonTapped), for: .touchUpInside)
```
Comment (JakubJankowski) [inline Customer Demo/Customer Demo/Screens/SignIn/SignInViewController.swift:24]: Out of curiosity (not a change request) - why didn't you handle it using the `@IBAction`?

**Review by JakubJankowski — APPROVED** [review]
LGTM 👍 
Especially nice to see a URLSession/URLRequest solution after staring at Alamofire in almost every other project 😁

File: Customer Demo/Customer Demo/Screens/SignIn/SignInViewController.swift (hunk 4 above)
Comment (arturjaworski) [reply Customer Demo/Customer Demo/Screens/SignIn/SignInViewController.swift:24]: it's clearer for me and easier to find out how it works

**Pull request merged**